src/utils/apiproto_utils/src/apiproto.c

```diff
@@ -232,34 +256,68 @@ ssize_t msg_send_resp(int fd, const ApiWireResponseMsg* msg)
 
 ssize_t msg_recv_resp(int fd, ApiWireResponseMsg* out_msg)
 {
-    ssize_t total_bytes_read = 0, bytes_read = -1, wait_res = -1;
+    ssize_t total_bytes_read = 0, bytes_read = -1;
     uint16_t msg_code = 0, msg_ret_val = 0;
     char read_buf[RESP_MSG_READ_BUF_SIZE] = { 0 };
+    int retries = 0;
+    const int MAX_RETRIES = 20; // ~3 seconds total with 150ms sleeps
 
-    while (total_bytes_read < (ssize_t)(RESP_MSG_READ_BUF_SIZE))
+    // Single select call with longer timeout for entire message
+    fd_set read_fds;
+    struct timeval timeout;
+    FD_ZERO(&read_fds);
+    FD_SET(fd, &read_fds);
+    timeout.tv_sec = 3; // 3 second timeout
+    timeout.tv_usec = 0;
+
+    int rdy = select(fd + 1, &read_fds, NULL, NULL, &timeout);
+    if (rdy <= 0)
     {
-        if ((wait_res = _wait_until_bytes_avail_for_read(fd)) < 0)
+        if (rdy == 0)
         {
-            return wait_res;
+            Log_Warn("msg_recv_resp: timeout waiting for response");
+            return MSGREV_AGAIN;
         }
+        Log_Error("msg_recv_resp: select error: %d (%s)", errno, strerror(errno));
+        return -1;
+    }
 
+    // Now read without select in loop - data is available
+    while (total_bytes_read < (ssize_t)(RESP_MSG_READ_BUF_SIZE) && retries < MAX_RETRIES)
+    {
         bytes_read = read(fd, read_buf + total_bytes_read, RESP_MSG_READ_BUF_SIZE - total_bytes_read);
         if (bytes_read < 0)
         {
-            if (errno == EINTR)
+            if (errno == EINTR || errno == EAGAIN || errno == EWOULDBLOCK)
             {
+                retries++;
+                usleep(150000); // 150ms
                 continue;
             }
             Log_Error("msg_recv_resp: read error: %d (%s)", errno, strerror(errno));
             return -1;
         }
         if (bytes_read == 0)
         {
-            break; // EOF
+            // EOF - writer closed before we got all data
+            if (total_bytes_read == 0)
+            {
+                // No data yet, retry a few times for slow ARM systems
+                retries++;
+                usleep(150000); // 150ms
```

Comment:
Is it possible for the writer to write something, wait for 50ms, then close the pipe? before this 150ms sleep is over?